Makefile

```diff
@@ -64,13 +64,15 @@ go-deps:
 	chmod +x ./vendor/k8s.io/code-generator/generate-groups.sh
 	chmod +x ./vendor/k8s.io/code-generator/generate-internal-groups.sh
 
+
+# Always install a version of setup-envtest that matches the MCO's kube version
 SETUP_ENVTEST := $(shell command -v setup-envtest 2> /dev/null)
 install-setup-envtest:
 ifdef SETUP_ENVTEST
 	@echo "Found setup-envtest"
 else
 	@echo "Installing setup-envtest"
-	go install -mod= sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+	go install -mod= sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240315194348-5aaf1190f880
```

Comment:
Let's add a comment so that we remember to update it instead of carrying over same version in future release

Reply:
I Should have checked https://github.com/kubernetes-sigs/controller-runtime/tree/main/tools/setup-envtest repo earlier . It looks like this repo doesn't have each kube version specific branches. So, let's remove the comment (sorry, should have checked earlier)
As per https://github.com/kubernetes-sigs/controller-runtime/commit/4c2442e4d743d172a2e0126a841ecd274fffc228 Go bump to 1.22 was added because kube 1.30 needs that. We can backport this PR to older version as it is and create a story targeted to 4.17 where we would bump kube to 1.30 to use latest tag.